Use parent class attribute  supports_out for foreach_zero opinfo

isuruf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112781
* __->__ #112778

Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

isuruf committed:
Use parent class attribute  supports_out for foreach_zero opinfo

Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

lezcano commented on test/test_foreach.py:
```diff
@@ -817,9 +817,9 @@ def test_foreach_copy_with_multi_device_inputs(self, device, dtype, op):
+        if (not inplace) and not op.supports_out:
             self.skipTest("out-of-place not implemented")
-        if inplace and op.has_no_in_place:
+        if inplace and not op.has_no_in_place:
```
Is this change correct? The name of the arg `has_no_in_place` is just not descriptive at all... ugh

isuruf replied:
Reverted.

lezcano reviewed (approved):
Just one comment, otherwise lgtm.

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

lezcano commented:
this PR is already approved

isuruf commented:
@pytorchbot merge

isuruf committed:
Update on "Use parent class attribute  supports_out for foreach_zero opinfo"


Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

[ghstack-poisoned]

isuruf commented:
@pytorchbot merge
